internal/k8s/controller.go

```diff
@@ -246,11 +247,14 @@ func NewLoadBalancerController(input NewLoadBalancerControllerInput) *LoadBalanc
 
 	lbc.syncQueue = newTaskQueue(lbc.sync)
 	if input.SpireAgentAddress != "" {
-		var err error
-		lbc.spiffeCertFetcher, err = NewSpiffeCertFetcher(lbc.syncSVIDRotation, input.SpireAgentAddress)
+		client, err := workloadapi.New(lbc.ctx, workloadapi.WithAddr("unix://"+input.SpireAgentAddress))
```

Comment:
What if our library just did this initialization in the constructor of the CertFetcher?

Reply:
Then we could likely remove NIC's dependency on the spiffe library altogether.

Reply:
Unfortunately the cert synchronization callback still requires workloadapi for the argument type. Would you like me to typedef that in our library? then we can remove it.

Reply:
No that's okay, I didn't realize it was used elsewhere.